Complete Authflow with WorkOS and SpaceTimeDB

ImGajeed76 opened the pull request:
(to tired to write the description, I'm sure CodeRabbit will do a good job at describing it 💀)

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Adds web dashboard and Tauri desktop app with user management, device-based OAuth (authorize/poll/token), account setup/unlock/recover, and a dashboard showing user info.
  * Client/server token flows, background connection to realtime DB, and vault-based key storage for secure crypto workflows.

* **UX / Design**
  * New cohesive theme, persistent light/dark toggle, PWA manifest, and copy-to-clipboard helper.

* **Documentation**
  * Added DEVELOPING.md and updated README with setup and quick-start.

* **Chores**
  * Added project configs: formatting, linting, tooling, package manifests, and ignore files.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

ImGajeed76 committed:
feat: initial dashboard commit

ImGajeed76 committed:
feat: simple login flow

ImGajeed76 committed:
feat: custom jwt generation

ImGajeed76 committed:
feat: added public jwks endpoint

ImGajeed76 committed:
feat: implemented user creation in spacetimedb

ImGajeed76 committed:
feat: improved spacetimedb security

ImGajeed76 committed:
Squashed 'SpacetimeDB/' content from commit 1fb0e102

git-subtree-dir: SpacetimeDB
git-subtree-split: 1fb0e10244656ca9681aedc6512017eabb72c68e

ImGajeed76 committed:
Merge commit 'c0881a763a1658d5ac001fc6bf6ca944ab9099cd' as 'SpacetimeDB'

ImGajeed76 committed:
feat: added scripts for setup and styleguide

ImGajeed76 committed:
feat: added scripts for setup and styleguide

ImGajeed76 committed:
fix: tsconfig and linting for .well-known

ImGajeed76 committed:
feat: improved setup screen and master password persistency

ImGajeed76 committed:
feat: added unlock and recovery screen

ImGajeed76 committed:
feat: moved from server side spacetime to client side

ImGajeed76 committed:
fix: rust fmt

ImGajeed76 committed:
feat: added signing check for updating keys

ImGajeed76 committed:
feat: new theme and updated env variable

ImGajeed76 committed:
Squashed 'SpacetimeDB/' changes from 1fb0e102..b9bbe793

b9bbe793 Merge pull request #2 from clockworklabs/master
c3cfadb2 SATS: Move some method bodies to be provided in 'Serializer' instead (#2921)
6a8cd7e9 Extract `object_pool` module from `PagePool` (#2920)
1166d68e Extract `spacetimedb-memory-usage` & Nix `bindings` dep on `spacetimedb-data-structures` (#2919)
6f5e33ab `update_database`: `is_unique` should only use the table's constraints (#2915)
06ba5293 Remove unused `get_column_by_field` (#2917)
ef0b2569 core: Allow caller to decide whether or not IRQ cores should be reserved (#2913)
c8716106 Record transaction metrics off the main thread (#2910)
e4d5c18f Wrap websocket flush in timeout (#2908)
29a0df42 `tools/upgrade-version` updates the Change Date in `LICENSE.txt` (#2895)
9aebe31f Filter out dropped clients in the send worker (#2899)
a7293a3c (perf) `send_one_computed_queries`: reuse aggregation `HashMap`s (#2909)
0c363518 Auto-migrate: Allow adding new variants at the tail (#2874)
47fd1684 durability: Fix task leak (#2875)
56f1f605 core: Allow adjusting CPU reservation pools (#2907)
022ca683 Fix query overwrites in the subscription manager (#2905)
28186d87 Track disconnects initiated by the client (#2893)
c3afc171 Remove unnecessary `.clone()` in `pipelined.rs` (#2897)
67703b1f Bump C# versions to 1.2.1 (#2890)
bc4fea20 `TypedIndex`: add special cases for `F{32,64}` (#2885)
971ae75b Improve serialization speed of enums without data in C# (#2762)
2f1d99ce Log when a websocket connection has ended (#2883)
9608f498 Speed up metadata update on unsubscribe (#2880)
8d250098 Fix two places we grab blocking locks on tokio runtime worker threads (#2881)
718712a8 Record the size of a client's outgoing message queue on drop (#2877)
b63df787 Track metadata updates for unsubscribes (#2876)
4f493e4d Relax conditions for query pruning (#2873)
1318e7e1 `messages::serialize`: take/put buffers from/into a `SerializeBufferPool` (#2823)
27af02ce Use database identity not owner identity for metrics (#2868)
da0757b5 Remove incorrect `const` qualifiers on two imports in `bindings.c` (#2862)
9f03b2c6 Fix an outdated docstring on `identity` and `try_identity` in the Rust SDK (#2871)
c522c0f9 Prune more queries when evaluating subscription updates (#2855)
70df4fb0 Test subscription updates for dml (#2716)
228c761c Use a scope guard for decrementing queue length. (#2865)
1e3e8db9 Database-specific `Timestamp` route (#2864)
193c0d48 Add more subscription metrics (#2861)
2d756dbc Wrap `try_init_host` futures in `tokio::spawn` so they don't get cancelled (#2857)
f4f698ef Always use identity when hashing queries from owners (#2859)
013e268e Fixed rename by using put instead of post (#2764)
12abd48a Send one-off responses through the broadcast queue. (#2853)
053fc6d9 Tidy up ModuleKind conversions a tiny bit (#2840)
4974676c client-api: Don't debug-print dropped outgoing Ws messages at info (#2819)
a19bd458 Reserve cores 0 and 1 for the OS (#2851)
0866fd39 Add an HTTP route to get the current `Timestamp` (#2850)
dc220ff3 fix: rust fmt

git-subtree-dir: SpacetimeDB
git-subtree-split: b9bbe793f39d88207a455c612f55f348d6cb44b3

ImGajeed76 committed:
Merge commit 'b2ea16f2de087b0718db731e961bf005d8993c47' into feat/dashboard-auth-workos

ImGajeed76 committed:
feat: first approach to new cli

ImGajeed76 committed:
fix: updated name to AxonotesCore

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/axonotes/AxonotesCore?pullRequest=27) <br/>All committers have signed the CLA.

ImGajeed76 committed:
feat: organized server structure

ImGajeed76 committed:
fix: updated formatting

ImGajeed76 committed:
feat: implemented desktop auth flow

ImGajeed76 committed:
feat: initial tauri setup

ImGajeed76 committed:
feat: tailwindcss setup

ImGajeed76 committed:
feat: initial tauri setup

ImGajeed76 committed:
feat: simple working auth flow

ImGajeed76 committed:
feat: implemented oauth pkce flow instead of deeplink

this flow is not completely finished yet!

ImGajeed76 committed:
feat: implemented oauth pkce flow instead of deeplink

this flow is not completely finished yet!

ImGajeed76 committed:
feat: removed old rust cli

ImGajeed76 committed:
feat: NEW CONFIG SYSTEM

from now on all config will be defined in the global axonotes.config.ts file. new config files or variables can be just added and after running `bun run config:generate` it will update it in all places.

todo:
- watch doesnt work correctly yet
- improve env variables in the dashboard

ImGajeed76 committed:
feat: formatted code

ImGajeed76 committed:
feat: using now axogen

ImGajeed76 committed:
feat: migrated to axogen

ImGajeed76 committed:
feat: removed SpacetimeDB subtree

ImGajeed76 committed:
feat: simple auth flow in tauri app

ImGajeed76 committed:
feat: tailwindcss + issue fix + refresh desktop

- Tailwindcss is now available in the desktop frontend
- There were typescript problems with Uint8Array and Uint8Array<ArrayBuffer>.
- Tokens are now generated based on the client_type attribute.

ImGajeed76 committed:
feat: desktop refresh + multi user

- The desktop app can now refresh its tokens in the background
- The desktop app can now add, update and remove users

ImGajeed76 committed:
feat: changed to zod and little cleanup

ImGajeed76 commented:
@coderabbitai full review

ImGajeed76 committed:
fix: coderabbit suggestions

- Remove duplicate permission and fix missing schema path
- Fix lint-staged glob for Tauri Rust sources (subdirectories not covered).
- Add OAuth2-recommended no-store headers (and Retry-After on 429).
- Do not upcast client_type from refresh token to desktop.
- Normalize user_code input (strip all dashes, uppercase) to reduce UX friction.
- Apply the same normalization in validateUserCode and update regex.

ImGajeed76 committed:
format: server.ts

ImGajeed76 commented:
@coderabbitai generate docstrings
